applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/explicit_central_differences_scheme.hpp

```diff
@@ -355,33 +361,40 @@ class ExplicitCentralDifferencesScheme
     /**
      * @brief This method initializes some rutines related with the explicit scheme
      * @param rModelPart The model of the problem to solve
+     * @param DoF The current dimention of the problem
      */
-    void InitializeExplicitScheme(ModelPart& rModelPart)
+    void InitializeExplicitScheme(
+        ModelPart& rModelPart,
+        const SizeType DoF = 3
+        )
     {
         KRATOS_TRY
 
         /// The array of ndoes
         NodesArrayType& r_nodes = rModelPart.Nodes();
 
-        /// Working in 3D always by default
-        const SizeType dim(3);
-
         // The first iterator of the array of nodes
-        auto it_begin = rModelPart.NodesBegin();
+        const auto it_begin = rModelPart.NodesBegin();
 
         /// Initialise the database of the nodes
-        const array_1d<double, 3> zero_array(3, 0.0);
-        #pragma omp parallel for firstprivate(it_begin)
+        const array_1d<double, 3> zero_array = ZeroVector(3);
+        #pragma omp parallel for
         for (int i = 0; i < static_cast<int>(r_nodes.size()); ++i) {
             auto it_node = (it_begin + i);
             it_node->SetValue(NODAL_MASS, 0.0);
             it_node->SetValue(MIDDLE_VELOCITY, zero_array);
-            it_node->SetValue(MIDDLE_ANGULAR_VELOCITY, zero_array);
-            it_node->SetValue(NODAL_INERTIA, zero_array);
         }
-
         const bool has_dof_for_rot_z = it_begin->HasDofFor(ROTATION_Z);
-        #pragma omp parallel for firstprivate(it_begin)
+        if (has_dof_for_rot_z) {
+            #pragma omp parallel for
+            for (int i = 0; i < static_cast<int>(r_nodes.size()); ++i) {
+                auto it_node = (it_begin + i);
+                it_node->SetValue(MIDDLE_ANGULAR_VELOCITY, zero_array);
+                it_node->SetValue(NODAL_INERTIA, zero_array);
+            }
+        }
+
+        #pragma omp parallel for
```

Comment:
missing firstprivate (if you don't put it you have false sharing and hence worst performance)

Reply:
Really?, if shared is const is not needed

Reply:
clearifying this will help us a lot with other code
I also was wondering what is better all the time

Reply:
maybe also @roigcarlo @msandre @jcotela know sth

Reply:
ok, to me the link and explanation you posted looks convincing. To me this conversation can be resolved (i am not doing it in case @roigcarlo @msandre or @jcotela want to comment)